db/1.14/modules/ROOT/pages/database-connector-troubleshooting.adoc

```diff
@@ -0,0 +1,299 @@
+. Apply the changes.
+
+== Troubleshoot Pooling Profiles Connections Making the Flow Hang
+Using a pooling profile that provides fewer connections than the flow requires can make the entire flow hang.
+
+To resolve this issue:
+
+One solution is to increase the size of the pool. When that's not possible, use Anypoint Connector for Virtual Machine (VM Connector) to shorten the life of the connection to match the life of a VM message. This way, every call to the *Stored procedure* operation is executed, its result is consumed, and the connection associated with the operation is closed and returned to the pool.
+
+The following example shows how to use pooling profiles with the *Stored procedure* operation and VM Connector. In this example, When the `foreach` element reaches the eleventh iteration, the *Stored procedure* operation doesn't hang, because the pool is able to provide a connection:
+
+[source,xml,linenums]
+----
+	<db:config name="Database_Config">
+		<db:generic-connection url="someUrl" user="someUser" password="somePassword" driverClassName="someDriver">
+			<db:pooling-profile maxPoolSize="10">
+			</db:pooling-profile>
+		</db:generic-connection>
+	</db:config>
+
+	<vm:config name="VM_Config">
+		<vm:queues >
+			<vm:queue queueName="testQueue" />
+		</vm:queues>
+	</vm:config>
+
+	<flow name="processingFlow">
+		<foreach collection="#[1 to 11]">
+			<logger level="INFO" message="#['Starting Iteration ' ++ payload as String]" />
+			<vm:publish config-ref="VM_Config" queueName="testQueue"/>
+			<logger level="INFO" message="#['Finishing Iteration ' ++ payload as String]"/>
+		</foreach>
+	</flow>
+
+	<flow name="coolFlow">
+		<vm:listener config-ref="VM_Config" queueName="testQueue" numberOfConsumers="1"/>
+		<db:stored-procedure config-ref="Database_Config">
+				<db:sql><![CDATA[{call testProc2()}]]></db:sql>
+			</db:stored-procedure>
+		<logger level="INFO" message="#[output application/json --- payload.resultSet1]" />
+	</flow>
+----
+
+== Understand Select and Stored Procedure Operations Warning Exceptions
+
+When the Database Connector *Select* and *Stored procedure* operations execute correctly but do not return results, Mule runtime engine (Mule) returns a `WARN` exception message instead of throwing the exception.
+
+For the *Select* operation, the behavior occurs when executing an invalid SQL statement, and for the *Store procedure* operation, when there is an error within a set of statements that can collapse the procedure. In fact, JDBC returns a series of `ResultSets`, `UpdateCounts`, and `Exceptions`.
+
+Mule does not throw an exception for the flow to catch it because the root cause is in the driver's behavior. Some drivers comply strictly with JDBC and properly throw an exception at execution time, but other drivers do not. Therefore, adding in the connector code the behavior of throwing an exception would break backward compatibility for Mule apps that do not expect an exception.
```

Comment:
I can't tell if this is referring to both operations or only to the second one.

Reply:
This refers to both operations.